docs: add message on click of hero in toh-4

ajitsinghkaler opened the pull request:
First we used a messages array to display messages toh-4 but it displayed only one message added additional messages on click of a hero

Fixes #28739

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #28739


## What is the new behavior?
Now on click of each hero additional message is there

https://angular-ubavw5.stackblitz.io can also be checked here

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

ajitsinghkaler commented:
@kapunahelewong I'm a little confused on how to add this in the docs flow. Can you please help me a little.

kapunahelewong commented:
Hi @ajitsinghkaler! You're on the right track. Yes, we should add a test for it. Are you comfortable writing it? If not, I can help. I'd like to get @jbogarthyde on this to help review from a tech writing perspective. I'll request her review. Thanks a lot for this, @ajitsinghkaler 🌟

ajitsinghkaler commented:
@kapunahelewong I'll try to write a test after that you can review

kapunahelewong commented:
Very nice, @ajitsinghkaler. As an example to work from, you could refer to this one: https://github.com/angular/angular/blob/master/aio/content/examples/getting-started/e2e/src/app.e2e-spec.ts#L52

But there are many in the example apps you could use as a model (or you could do it your own way). You'll just want to be sure to test that the number and text are as you expect them to be on click. Let me know if you have any questions. Thank you very much!

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -19,7 +19,7 @@ Services are a great way to share information among classes that _don't know eac
 You'll create a `MessageService` and inject it in two places:
```
end with period (.), not colon (:), per Google style.

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -19,7 +19,7 @@ Services are a great way to share information among classes that _don't know eac
 You'll create a `MessageService` and inject it in two places:
 
 1. in `HeroService` which uses the service to send a message
```
1. Inject in `HeroService`, which uses the service to send a message.
(complete sentence, end with period)

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -19,7 +19,7 @@ Services are a great way to share information among classes that _don't know eac
 You'll create a `MessageService` and inject it in two places:
 
 1. in `HeroService` which uses the service to send a message
-2. in `MessagesComponent` which displays that message
```
1. Inject in `MessagesComponent` which displays that message, and also displays the ID when the user clicks a hero.

(you can use "1." for all steps, the parser automatically numbers them correctly)

mary-poppins commented:
You can preview 92a4df0 at https://pr34496-92a4df0.ngbuilds.io/.

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -367,9 +367,21 @@ to `MessageService.clear()`.
+</code-example>
+
+Here on click of each hero we are adding a message to the message area. To get a much clear picture of how to use the message service and adding additional messages.
+
```
remove this paragraph

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -367,9 +367,21 @@ to `MessageService.clear()`.
 Scroll to the bottom to see the message from the `HeroService` in the message area.
-Click the "clear" button and the message area disappears.
+Click the "clear" button and the message area disappears. On clicking of heroes you get a
+message selected hero id=heroid.
```
(replace the whole paragraph with the following)
Refresh the browser to see the list of heroes, and scroll to the bottom to see the messages from the `HeroService`.
Each time you click a hero, a new message appears to record the selection. Use the "clear" button to clear the message history.

jbogarthyde reviewed (changes requested):
Edited the descriptive text.

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -367,9 +367,21 @@ to `MessageService.clear()`.
+
+We can add additional messages to hero service on click of eachhero which will help us in
+routing in the next tutorials.
+
```
Replace 372-373 with:

The following example shows how to send and display a message each time the user clicks on a hero, showing a history of the user's selections. This will be helpful when you get to the next section on [Routing](tutorial/toh-pt5).

mary-poppins commented:
You can preview 281c6b9 at https://pr34496-281c6b9.ngbuilds.io/.

ajitsinghkaler commented:
@jbogarthyde @kapunahelewong  I've made the changes  can you please have a look

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -367,9 +368,20 @@ to `MessageService.clear()`.
+
+The following example shows how to send and display a message each time the user clicks on
+a hero, showing a history of the user's selections. This will be helpful when you get to the
+next section on Routing.
```
Please add the link on Routing.

ajitsinghkaler replied:
We should add a link to routing or to toh-5

jbogarthyde replied:
To Toh-5

jbogarthyde replied:
To Toh-5

jbogarthyde commented on aio/content/tutorial/toh-pt4.md:
```diff
@@ -16,10 +16,11 @@ you'll rely on Angular [*dependency injection*](guide/dependency-injection)
-1. in `HeroService` which uses the service to send a message
-2. in `MessagesComponent` which displays that message
+1. Inject in HeroService, which uses the service to send a message.
+2. Inject in MessagesComponent which displays that message, and also displays the ID
```
Add comma before “which”

jbogarthyde reviewed (changes requested):
A couple of small changes.

ajitsinghkaler commented:
@jbogarthydemade the changes

mary-poppins commented:
You can preview de928b8 at https://pr34496-de928b8.ngbuilds.io/.

jbogarthyde commented:
Routing is toh5, not toh6.

ajitsinghkaler committed:
docs: add message on click of hero in toh-4

First we used a messages array to display messages toh-4 but it dsplayed only one message added additional messages on click of a hero

Fixes #28739

jbogarthyde reviewed (approved):
Thanks - looks good😊

mary-poppins commented:
You can preview d6c9ea2 at https://pr34496-d6c9ea2.ngbuilds.io/.

ajitsinghkaler commented:
@jbogarthyde Do we need to add anything else? In like toh-5 because the previous content it will be asking will be changed

kapunahelewong commented:
@ajitsinghkaler thank you for the consideration but I have to revisit my PR on it anyway, so whatever you get merged I'll just rebase off of. 🙂

ajitsinghkaler commented:
@kapunahelewong can you please help me in creating tests ou were talking about

kapunahelewong commented:
Happily, @ajitsinghkaler! I've started the e2e test edits, and I think it's pretty close, but I have to go to an appointment! Anyway, look at [lines 135 and 136 in the e2e for toh-pt4](https://github.com/angular/angular/pull/34496/files#diff-ceadbec94be0c25ce7eb0fe2e9ad4df3R136). That line isn't selecting correctly and needs to be fixed. I'm using it on [line 90](https://github.com/angular/angular/pull/34496/files#diff-ceadbec94be0c25ce7eb0fe2e9ad4df3R136). To run the tests, do:

```yarn example-e2e --filter toh-pt4```

Comment out my changes to see everything pass. I hope this helps. I can help more later!

ajitsinghkaler commented:
@kapunahelewong I think this is because at the first tie when no hero is clicked then there is no message in the dom first we have to create a virtual click in the dom to check this functionality.

ajitsinghkaler commented:
@kapunahelewong the tests are failing even if I comment out your code

ajitsinghkaler commented:
@kapunahelewong I've made the changes

ajitsinghkaler commented:
I messed up rebasing.

Sorry for wrongly sending a request review to all.

ajitsinghkaler commented:
@kapunahelewong I think I added all the tests I'm trying to rebase it should I squash your and my commits into one

kapunahelewong commented:
@ajitsinghkaler, great! Yes, please do squash all our commits into one. If it messes up, let me know and I can look at it from my end. Thank you!

mary-poppins commented:
You can preview 88d8f58 at https://pr34496-88d8f58.ngbuilds.io/.

ajitsinghkaler commented:
@kapunahelewong made the changes please have a look

mary-poppins commented:
You can preview 15ecc03 at https://pr34496-15ecc03.ngbuilds.io/.

kapunahelewong reviewed (approved):
LGTM. I've squashed those last two commits and amended the commit message. Let's make sure it passes the tests. Otherwise, everything appears to be in order. Thank you @ajitsinghkaler!

mary-poppins commented:
You can preview e1c722c at https://pr34496-e1c722c.ngbuilds.io/.

kapunahelewong commented:
merge-assistance: Global approval for this docs-only change.
